### ISSUE TITLE
For thaumcraft 5(possibly 4 if you have the time and patience to edit all your (tile)netities)

### ISSUE THREAD

**Issue by tschallacka** [Azanor/thaumcraft#1565]
Regarding my extremely vague post on the minecraft forums ;-)

On my server where I run only thaumcraft all my players who play via wifi have issues with corrupted data and get d/cd like at least once every 30 minutes with internal server errors and such like that.

What happens is their wifi signal gets disrupted by a microwave, a cellphone, something else with a decent EMP field, and thusly corrupting some nbt data being transferred or the packets of your custom entities(golems etc..) being impaired.

This is especially bad in tainted biomes. Some players need to be tpd out of tainted biomes because they crash every 10 seconds to 60 seconds in them.
I advice them to change wifi channels etc or use a cabled connection to get a stable connection with less corruption.

If you try/catch your readNBT/readCustomNBT methods and the packet handlers for your mobs a lot of these errors won't crash them out of the server but can be handled gracefully.

Basically data on the internet cannot be guaranteed to arrive safely and a simple safety net should prevent these crashes due to invalid variables being tried to set.

I hope this is less vague :P

**Comment by chrisbecke** [comment]
Actually more confused. The corruption of which you speak should not be possible. tcp does rudimentary packet checksumming which should cause the connection to terminate regardless. Even if the errors were bypassing that protection due to badly written drivers, the corruption would be anywhere in the packet and a crash will be immanent anyhow. This is totally not a worthwhile scenario to guard against.

**Comment by forsaken1111** [comment]
This type of data corruption you are talking about is handled at the transport/session layers in the OSI model, not by applications. There isn't much thaumcraft could do to guard against this realistically.

**Comment by tschallacka** [comment]
Please do tell me again about your network driver wrappers and error checking in packets.

https://gist.github.com/tschallacka/cbf12220c1ec90ed8b70

In this case its the runic packets, I had also have these with the golem packets, nbt data of  tc mobs etc...

**Comment by Azanor** [comment]
There is really nothing I can do about data corruption in packets - that is all handled internally by the network code. 

HOWEVER, if you had posted the crashlog in the first place, instead of assuming what and where the problem is I could have told you that the error actually has very little to do with corrupted packets or something like that.

**Comment by mSt3in** [comment]
@Azanor close?
